### PR TITLE
feat: Rename settlement price decimals to settlement data decimals

### LINKED ISSUE #5641: Reconsider naming of "settlement data" endpoint in APIs
# Problem encountered
The APIs reference something called 'settlement price decimals', however this won't scale to non-cash products and settlement data is not necessarily a price, even for cash settled futures. 

Suggestion: rename to settlementDataDecimals in line with the [product parameter](https://github.com/vegaprotocol/specs/blob/master/protocol/0016-PFUT-product_builtin_future.md#1-product-parameters). 

# Observed behaviour
Example: https://docs.vega.xyz/docs/testnet/grpc/vega/markets.proto#future

# Definition of Done
>ℹ️ Not every issue will need every item checked, however, every item on this list should be properly considered and actioned to meet the [DoD](https://github.com/vegaprotocol/vega/blob/develop/DEFINITION_OF_DONE.md).

**Before Merging**
- [ ] Code refactored to meet SOLID and other code design principles
- [ ] Code is compilation error, warning, and hint free
- [ ] Carry out a basic happy path end-to-end check of the new code
- [ ] All APIs are documented so auto-generated documentation is created
- [ ] All bug recreation steps can be followed without presenting the original error/bug
- [ ] All Unit, Integration and BVT tests are passing
- [ ] Implementation is peer reviewed (coding standards, meeting acceptance criteria, code/design quality)
- [ ] Create [front end](https://github.com/vegaprotocol/token-frontend/issues) or [console](https://github.com/vegaprotocol/console/issues) tickets with feature labels (should be done when starting the work if dependencies known i.e. API changes)

**After Merging**
- [ ] Move development ticket to `Done` if there is **NO** requirement for new [system-tests](https://github.com/vegaprotocol/system-tests/issues)
- [ ] Resolve any issues with broken system-tests
- [ ] Create [documentation](https://github.com/vegaprotocol/documentation/issues) tickets with feature labels if functionality has changed, or is a new feature

### PR DESCRIPTION
**Pull request by ze97286**
Closes #5641

### CODE REVIEW

**Review by jeremyletang — DISMISSED** [review]

**Review by jeremyletang — APPROVED** [review]

**Review by mattrussell36 — APPROVED** [review]

File: datanode/gateway/graphql/schema.graphql [hunk 1]
```diff
@@ -1340,8 +1340,8 @@ type Future {
   "The binding between the oracle spec and the settlement price"
   oracleSpecBinding: OracleSpecToFutureBinding!
 
-  "The number of decimal places implied by the settlement price emitted by the settlement oracle"
-  settlementPriceDecimals: Int!
+  "The number of decimal places implied by the settlement data emitted by the settlement oracle"
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:1343]:
```suggestion
  "The number of decimal places implied by the settlement data (such as price) emitted by the settlement oracle"
```
suggestion for an example of data for guidance

File: datanode/gateway/graphql/schema.graphql [hunk 2]
```diff
@@ -3110,8 +3110,8 @@ type FutureProduct {
   """
   oracleSpecBinding: OracleSpecToFutureBinding!
 
-  "The number of decimal places implied by the settlement price emitted by the settlement oracle"
-  settlementPriceDecimals: Int!
+  "The number of decimal places implied by the settlement data emitted by the settlement oracle"
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:3113]:
```suggestion
  "The number of decimal places implied by the settlement data (such as price) emitted by the settlement oracle"
```

File: protos/sources/vega/governance.proto [hunk 3]
```diff
@@ -20,8 +20,8 @@ message FutureProduct {
   oracles.v1.OracleSpecConfiguration oracle_spec_for_trading_termination = 6;
   // The binding between the oracle spec and the settlement price
   OracleSpecToFutureBinding oracle_spec_binding = 7;
-  // The number of decimal places implied by the settlement price emitted by the settlement oracle
-  uint32 settlement_price_decimals = 8;
+  // The number of decimal places implied by the settlement data emitted by the settlement oracle
```
Comment (candida-d) [inline protos/sources/vega/governance.proto:23]:
```suggestion
  // The number of decimal places implied by the settlement data (such as price) emitted by the settlement oracle
```

File: protos/sources/vega/governance.proto [hunk 4]
```diff
@@ -114,8 +114,8 @@ message UpdateFutureProduct {
   oracles.v1.OracleSpecConfiguration oracle_spec_for_trading_termination = 3;
   // The binding between the oracle spec and the settlement price
   OracleSpecToFutureBinding oracle_spec_binding = 4;
-  // The number of decimal places implied by the settlement price emitted by the settlement oracle
-  uint32 settlement_price_decimals = 5;
+  // The number of decimal places implied by the settlement data emitted by the settlement oracle
```
Comment (candida-d) [inline protos/sources/vega/governance.proto:117]:
```suggestion
  // The number of decimal places implied by the settlement data (such as price) emitted by the settlement oracle
```

File: protos/sources/vega/markets.proto [hunk 5]
```diff
@@ -34,8 +34,8 @@ message Future {
   // The binding between the oracle spec and the oracle data
   OracleSpecToFutureBinding oracle_spec_binding = 7;
 
-  // the number of decimal places implied by the settlement price emitted by the settlement oracle
-  uint32 settlement_price_decimals = 8;
+  // the number of decimal places implied by the settlement data emitted by the settlement oracle
```
Comment (candida-d) [inline protos/sources/vega/markets.proto:37]:
```suggestion
  // the number of decimal places implied by the settlement data (such as price) emitted by the settlement oracle
```

**Review by candida-d — APPROVED** [review]
Made a non-blocking suggestion for the descriptions to use price as an example of data

**Review by witgaw — APPROVED** [review]

**Pull request merged**